xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -4494,23 +4494,42 @@ int CVideoPlayer::AddSubtitleFile(const std::string& filename, const std::string
 {
   std::string ext = URIUtils::GetExtension(filename);
   std::string vobsubfile = subfilename;
-  if (ext == ".idx")
+  if (ext == ".idx" || ext == ".sup")
   {
-    if (vobsubfile.empty()) {
-      // find corresponding .sub (e.g. in case of manually selected .idx sub)
-      vobsubfile = CUtil::GetVobSubSubFromIdx(filename);
+    std::shared_ptr<CDVDDemux> pDemux;
+    if (ext == ".idx")
+    {
       if (vobsubfile.empty())
+      {
+        // find corresponding .sub (e.g. in case of manually selected .idx sub)
+        vobsubfile = CUtil::GetVobSubSubFromIdx(filename);
+        if (vobsubfile.empty())
+          return -1;
+      }
+      auto pDemuxVobsub = std::make_shared<CDVDDemuxVobsub>();
+      if (!pDemuxVobsub->Open(filename, STREAM_SOURCE_NONE, vobsubfile))
         return -1;
+      m_SelectionStreams.Update(NULL, pDemuxVobsub.get(), vobsubfile);
+      pDemux = pDemuxVobsub;
+    }
+    else
```

Comment:
even if it is intuitive i would add a small comment in-line to `else` similar to `// .sup file`